app/assets/javascripts/discourse/app/services/lightbox.js

```diff
@@ -0,0 +1,272 @@
+    const { items, startingIndex } = await processHTML({ container, selector });
+
+    this.appEvents.trigger(LIGHTBOX_APP_EVENT_NAMES.OPEN, {
+      items,
+      startingIndex,
+      callbacks: { ...this.callbacks },
+      options: { ...this.options },
+    });
+
+    this.lightboxIsOpen = true;
+  }
+
+  @bind
+  async closeLightbox() {
+    if (this.lightboxIsOpen) {
+      this.appEvents.trigger(LIGHTBOX_APP_EVENT_NAMES.CLOSE);
+      this.lightboxIsOpen = false;
+    }
+  }
+
+  async #setupLightboxes({ container, selector }) {
+    if (!container) {
+      throw new Error("Lightboxes require a container to be passed in");
+    }
+
+    if (container.length) {
+      // passed in a nodelist instead of an element
+      container = findNearestSharedParent(container);
+    }
+
+    const hasLightboxes = container.querySelector(selector);
+
+    if (!hasLightboxes) {
+      return;
+    }
+
+    const handlerOptions = {
+      capture: true,
+    };
+
+    function clickHandler(event) {
+      const isLightboxClick = event
+        .composedPath()
+        .find(
+          (element) =>
+            element.matches &&
+            (element.matches(selector) ||
+              element.matches("[data-lightbox-trigger]"))
+        );
+
+      if (!isLightboxClick) {
+        return;
+      }
+
+      event.preventDefault();
+
+      this.openLightbox({
+        container,
+        selector,
+      });
+
+      container.toggleAttribute(SELECTORS.DOCUMENT_LAST_FOCUSED_ELEMENT);
+    }
+
+    container.addEventListener(
+      "click",
+      clickHandler.bind(this),
+      handlerOptions
+    );
+
+    this.lightboxClickElements.push({
+      container,
+      clickHandler,
+      handlerOptions,
+    });
+  }
+
+  @bind
+  async setupLightboxes({ container, selector }) {
+    this.#setupLightboxes({ container, selector });
+  }
+
+  async #cleanupLightboxes() {
+    this.closeLightbox();
+
+    this.lightboxClickElements.forEach(
+      ({ container, clickHandler, handlerOptions }) => {
+        container.removeEventListener("click", clickHandler, handlerOptions);
+      }
+    );
+
+    this.lightboxClickElements = [];
+  }
+
+  @bind
+  async cleanupLightboxes() {
+    this.#cleanupLightboxes();
+  }
+
+  async #setupDocumentFocus() {
+    this.lastFocusedElement = document.activeElement;
+    document.activeElement.blur();
+    document.querySelector("[data-lightbox-close-button]")?.focus();
+
+    const focusableElements = document.querySelectorAll(
+      "[data-lightbox-element] button"
+    );
+
+    const firstFocusableElement = focusableElements[0];
+    const lastFocusableElement =
+      focusableElements[focusableElements.length - 1];
+
+    const focusTraps = document.querySelectorAll("[data-lightbox-focus-trap]");
+
+    const firstfocusTrap = focusTraps[0];
+    const lastfocusTrap = focusTraps[focusTraps.length - 1];
+
+    this.onFocus = ({ target }) => {
+      if (target === firstfocusTrap) {
```

Comment:
At this point in time you are not sure the elements still exist here, most likely yes but still it will be happening in a different context